src/Selector/Xpath/Manipulator.php

```diff
@@ -43,6 +48,28 @@ public function prepend($xpath, $prefix)
             $prefix = '('.$prefix.')';
         }
 
+        // Compute a safe pipe placeholder, not included in the xpath string.
+        $suffix = 0;
+        do {
+            $placeholder = self::PIPE_PLACEHOLDER_TEMPLATE.$suffix++;
+        } while (false !== strpos($xpath, $placeholder));
```

Comment:
This should have a test case where "****PIPE-CHAR-PLACEHOLDER-0" is in the string to make sure raising the number here works.